include/log++.h

```diff
@@ -617,21 +674,31 @@ class InternalCondLog : public InternalLog {
 };
 
 
-class LogPolicy {
+class LogPolicyBase {
  public:
   virtual void update() = 0;
   [[nodiscard]] virtual bool shouldLog() const = 0;
   virtual void onLog() {};
-  virtual ~LogPolicy() = default;
- protected:
-  explicit LogPolicy(int max) : max_(max) {}
+};
+
+template<typename T>
+class LogPolicy : public LogPolicyBase {
+public:
+  explicit LogPolicy(T max) : max_(max) {}
+protected:
+  T max_{0};
```

Comment:
can T always be initalized with 0? Probably makes sense as its supposed to be float/int, correct? Could add a static assert in the class to make sure it terminates with a good warning if it doesn't work.

Reply:
There is a `static_assert` defined for that in Line 815-819. The `InternalLogCount::update()` method calls `InternalLogCount::updatePolicy()`, which is the only method where classes that inherit `LogPolicy<T>` are constructed. Therefore, `T` is guarded by the `static_assert`. `T` always has to be a numeric type and therefore can always be initialized with 0.